fpmms/profitability.py

```diff
@@ -0,0 +1,631 @@
+    fpmmTrades["trader_address"] = fpmmTrades["trader_address"].str.lower().str.strip()
+
+    return fpmmTrades, tools
+
+
+def determine_market_status(trade, current_answer):
+    """Determine the market status of a trade."""
+    if current_answer is np.nan and time.time() >= trade["fpmm.openingTimestamp"]:
+        return MarketState.PENDING
+    elif current_answer == np.nan:
+        return MarketState.OPEN
+    elif trade["fpmm.isPendingArbitration"]:
+        return MarketState.ARBITRATING
+    elif time.time() < trade["fpmm.answerFinalizedTimestamp"]:
+        return MarketState.FINALIZING
+    return MarketState.CLOSED
+
+
+def analyse_trader(
+    trader_address: str, fpmmTrades: pd.DataFrame, tools: pd.DataFrame
+) -> pd.DataFrame:
+    """Analyse a trader's trades"""
+    # Filter trades and tools for the given trader
+    trades = fpmmTrades[fpmmTrades["trader_address"] == trader_address]
+    tools_usage = tools[tools["trader_address"] == trader_address]
+
+    # Prepare the DataFrame
+    trades_df = pd.DataFrame(columns=ALL_TRADES_STATS_DF_COLS)
+    if trades.empty:
+        return trades_df
+
+    # Fetch user's conditional tokens gc graph
+    try:
+        user_json = _query_conditional_tokens_gc_subgraph(trader_address)
+    except Exception as e:
+        print(f"Error fetching user data: {e}")
+        return trades_df
+
+    # Iterate over the trades
+    for i, trade in tqdm(trades.iterrows(), total=len(trades), desc="Analysing trades"):
+        try:
+            # Parsing and computing shared values
+            creation_timestamp_utc = datetime.datetime.fromtimestamp(
+                trade["creationTimestamp"], tz=datetime.timezone.utc
+            )
+            collateral_amount = wei_to_unit(float(trade["collateralAmount"]))
+            fee_amount = wei_to_unit(float(trade["feeAmount"]))
+            outcome_tokens_traded = wei_to_unit(float(trade["outcomeTokensTraded"]))
+            earnings, winner_trade = (0, False)
+            redemption = _is_redeemed(user_json, trade)
+            current_answer = trade["fpmm.currentAnswer"]
+
+            # Determine market status
+            market_status = determine_market_status(trade, current_answer)
+
+            # Skip non-closed markets
+            if market_status != MarketState.CLOSED:
+                print(
+                    f"Skipping trade {i} because market is not closed. Market Status: {market_status}"
+                )
+                continue
+            current_answer = convert_hex_to_int(current_answer)
+
+            # Compute invalidity
+            is_invalid = current_answer == INVALID_ANSWER
+
+            # Compute earnings and winner trade status
+            if is_invalid:
+                earnings = collateral_amount
+                winner_trade = False
+            elif trade["outcomeIndex"] == current_answer:
+                earnings = outcome_tokens_traded
+                winner_trade = True
```

Comment:
I think that the payout is not the same as the `outcomeTokensTraded`.

cc @jmoreira-valory

Reply:
For winer markets it is outcomesTokensTraded. This can be seen from this example transaction:
https://gnosisscan.io/tx/0x2429fe81e89b949c3e78fd5f451e54a9dce5914bb9da1e7177eca8701a254077
And inspected in the subgraph as follows:
```
{
  fpmmTrades(where: {creator: "0x175cf7d914b482f42e76df01219c8ae578953390"}) {
    outcomeTokensTraded
    fpmm {
      id
    }
  }
}
```